climada_petals/hazard/copernicus_forecast/handler.py

```diff
@@ -0,0 +1,560 @@
+            Defaults to the current directory ('.').
+        url (str): url to the CDS API. Defaults to None, in which case the url from /.cdsapirc
+            is used.
+        key (str): CDS API key to the CDS API. Defaults to None, in which case the key from
+            /.cdsapirc is used.
+
+        Note:
+        This method sets up logging and initializes the data directory for the instance.
+        """
+        logging.basicConfig(format='%(asctime)s | %(levelname)s : %(message)s', level=logging.INFO)
+        self.logger = logging.getLogger()
+        self.data_dir = data_dir
+        self.key = key
+        self.url = url
+    
+    @staticmethod
+    def _get_bounds_for_area_selection(area_selection, margin=0.2):
+        """
+        Determines the geographic bounds based on an area selection string.
+
+        Parameters:
+        area_selection (str): Specifies the area for data selection.
+        margin (float): Additional margin to be added to the bounds in degrees.
+
+        Returns:
+        list: A list of four floats representing the bounds [north, east, south, west].
+        """
+        if isinstance(area_selection, str):
+            if area_selection.lower() == "global":
+                return [90, -180, -90, 180]  # north, west, south, east
+        else:
+            # try if area was given in bounds
+            try: 
+                north, west, south, east = area_selection
+                lat_margin = margin * (north - south)
+                lon_margin = margin * (east - west)
+                north += lat_margin
+                east += lon_margin
+                south -= lat_margin
+                west -= lon_margin
+                return [north, west, south, east]
+            except:
+                pass
+
+            # check if countries are given 
+            combined_bounds = [-90, 180, 90, -180]
+            for iso in area_selection:
+                geo = get_country_geometries(iso).to_crs(epsg=4326)
+                bounds = geo.total_bounds
+                if np.any(np.isnan(bounds)):
+                    logging.warning(f"ISO code '{iso}' not recognized. " \
+                        "This region will not be included." )
+
+                min_lon, min_lat, max_lon, max_lat = bounds
+
+                lat_margin = margin * (max_lat - min_lat)
+                lon_margin = margin * (max_lon - min_lon)
+
+                combined_bounds[0] = max(combined_bounds[0], max_lat + lat_margin)
+                combined_bounds[1] = min(combined_bounds[1], min_lon - lon_margin)
+                combined_bounds[2] = min(combined_bounds[2], min_lat - lat_margin)
+                combined_bounds[3] = max(combined_bounds[3], max_lon + lon_margin)
+            
+            if combined_bounds == [-90, 180, 90, -180]:
+                return None
+            else:
+                return combined_bounds
+
+    def explain_index(self, tf_index):
```

Comment:
Please use either `indicator` or `index` consistently in all the code.

Reply:
I keep indicator